### PR TITLE
adding logic for restricting operator branches

### PR DESCRIPTION
**Pull request by ewolinetz**
manual cherry-pick of https://github.com/openshift/origin-aggregated-logging/pull/1951

### CODE REVIEW

File: hack/deploy-logging.sh [hunk 1]
```diff
@@ -168,15 +168,24 @@ get_cluster_version_maj_min() {
     fi
 }
 
+MASTER_RELEASE_VERSION=4.6
```
Comment (jcantrill) [inline hack/deploy-logging.sh:171]: I thought we moved away from needing to hardcode this here.  Isn't there a VAR that comes from CI that tells us the release stream?  @bparees   would it make more sense to place this in the Makefile?

Comment (bparees) [reply hack/deploy-logging.sh:171]: i'm not aware of one....at best there might be one that tells you the branch (e.g. release-4.6).

as for putting it in the makefile or here.....  it appears to be a default here, that could be overridden from the makefile (or other invoker of the script)?

Comment (ewolinetz) [reply hack/deploy-logging.sh:171]: i'll update it so we could override it in here

**Comment by jcantrill** [comment]
overrding bz label as this address CI improvements

**Pull request merged**